# 可恢复错误

Result<T, E> 是一个带泛型的枚举.

```rust
enum Result<T, E> {
   Ok(T),
   Err(E),
}
```

Result<T, E> 通常用于函数的返回值, 用以表明该次函数调用是成功或失败. 它描述了函数调用过程可能出现的错误. Result<T, E> 可能有两种结果之一:

- OK(T): 成功, 并且获取到 T
- Err(E): 错误, 并且获取到错误描述 E

例子: 使用标准库打开一个文件

```rust
fn main() {
    match std::fs::read("/tmp/foo") {
        Ok(data) => println!("{:?}", data),
        Err(err) => println!("{:?}", err),
    }
}
```
